packages/frontend/src/app.tsx

```diff
@@ -0,0 +1,330 @@
+      transitionInterpolator: new FlyToInterpolator({ speed: 2 }),
+      transitionDuration: 'auto',
+    });
+  }, []);
+
+  const handleViewportChange = (nextViewState: ViewStateChangeParameters<{ zoom: number }>): void => {
+    setViewState({ ...viewState, ...nextViewState.viewState });
+    // for higher zoom levels lower the viewport's zoom level by 1 to have a buffer around the query bounds
+    const viewportZoom = nextViewState.viewState.zoom <= ZOOM_OFFEST + 1 ? nextViewState.viewState.zoom : nextViewState.viewState.zoom - 1;
+    const webMercatorViewport = new WebMercatorViewport({ ...nextViewState.viewState, zoom: viewportZoom });
+    currentBounds = webMercatorViewport.getBounds();
+    overrideComparator = true;
+
+    const nextZoom = Math.floor(nextViewState.viewState.zoom);
+    if (currentZoomLevel === nextZoom) {
+      return;
+    }
+
+    currentZoomLevel = nextZoom;
+    // reset min-max values between zoom changes
+    if (statsTable.metric) {
+      setStatsTable((prevStatsTable) => {
+        const nextMetric = prevStatsTable.metric ? { ...prevStatsTable.metric, range: { ...INITIAL_MIN_MAX } } : undefined;
+        return { ...prevStatsTable, metric: nextMetric };
+      });
+    }
+  };
+
+  const handleOpenSidebar = (data: TileDetails[]): void => {
+    setSidebarData(data);
+    setIsSidebarOpen(true);
+  };
+
+  const handleCloseSidebar = (): void => {
+    setSidebarData([]);
+    setIsSidebarOpen(false);
+  };
+
+  useEffect(() => {
+    if (selectedKit === undefined) {
+      return;
+    }
+    const prevMaxStateRange = selectedKit[MAX_KIT_STATE_KEY];
+    const kitMetadata = kits.find((kit) => kit.name === selectedKit.name);
+    setSelectedKit(kitMetadata);
+
+    if (+prevMaxStateRange !== stateRange[1]) {
+      return;
+    }
+    setStateRange([stateRange[0], kitMetadata ? +kitMetadata[MAX_KIT_STATE_KEY] : DEFAULT_MAX_STATE]);
+  }, [kits]);
+
+  useEffect(() => {
+    if (selectedMetric === undefined) {
+      return;
+    }
+    setStatsTable((prevStatsTable) => {
+      const nextMetric = { ...selectedMetric, range: { ...INITIAL_MIN_MAX } };
+      return { ...prevStatsTable, metric: nextMetric };
+    });
+    overrideComparator = true;
+  }, [selectedMetric]);
+
+  useEffect(() => {
+    async function kitsFetch(): Promise<void> {
+      try {
+        const [result, duration] = await timerify<Awaited<ReturnType<typeof client.getKits>>, never[]>(client.getKits.bind(client));
+        kits = result;
+        setStatsTable((prevStatsTable) => {
+          const nextHttpStat = calcHttpStat(prevStatsTable.httpInvokes.kits, duration);
+          return { ...prevStatsTable, httpInvokes: { ...prevStatsTable.httpInvokes, kits: nextHttpStat } };
+        });
+      } catch (err) {
+        logger.error({ msg: 'error fetching kits', err });
+        enqueueSnackbar(JSON.stringify(err), { variant: 'error' });
+      }
+    }
+    void kitsFetch();
+    setIntervalAsync(kitsFetch, FETCH_KITS_INTERVAL);
```

Comment:
Personally I prefer CRON over intervals as it makes it more predictable.

Reply:
using an async function thus implementation is done with `set-interval-async`